plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/DeltaLakeTableMetadataScheduler.java

```diff
@@ -0,0 +1,263 @@
+            @MaxTableParameterLength int tableParameterLengthLimit,
+            DeltaLakeConfig config)
+    {
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.tableOperationsProvider = requireNonNull(tableOperationsProvider, "tableOperationsProvider is null");
+        this.tableParameterLengthLimit = tableParameterLengthLimit;
+        this.storeTableMetadataThreads = config.getStoreTableMetadataThreads();
+        requireNonNull(nodeManager, "nodeManager is null");
+        this.enabled = config.isStoreTableMetadataEnabled() && nodeManager.getCurrentNode().isCoordinator();
+    }
+
+    @Managed
+    public boolean isShutdown()
+    {
+        return scheduler.isShutdown();
+    }
+
+    public void putAll(Map<SchemaTableName, TableUpdateInfo> tableParameters)
+    {
+        if (scheduler.isShutdown()) {
+            log.debug("Scheduler is already shutdown, skipping the update: %s", tableParameters);
+            return;
+        }
+        updateInfos.putAll(tableParameters);
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        if (enabled) {
+            executor = storeTableMetadataThreads == 0 ? newDirectExecutorService() : newFixedThreadPool(storeTableMetadataThreads, threadsNamed("store-table-metadata-%s"));
+            scheduler = newSingleThreadScheduledExecutor(daemonThreadsNamed("store-table-metadata"));
+
+            scheduler.scheduleWithFixedDelay(() -> {
+                try {
+                    process();
+                }
+                catch (Throwable e) {
+                    log.warn(e, "Error storing table metadata");
+                }
+                try {
+                    checkFailedTasks();
+                }
+                catch (Throwable e) {
+                    log.warn(e, "Error canceling metadata update tasks");
+                }
+            }, 200, 1000, MILLISECONDS);
+        }
+    }
+
+    @VisibleForTesting
+    public void process()
+    {
+        List<Callable<Void>> tasks = new ArrayList<>();
+        synchronized (this) {
+            if (updateInfos.isEmpty()) {
+                return;
+            }
+
+            Map<SchemaTableName, TableUpdateInfo> updateTables = updateInfos.entrySet().stream()
+                    .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue, maxBy(comparing(TableUpdateInfo::version))));
+
+            log.debug("Processing %s table(s): %s", updateTables.size(), updateTables.keySet());
+            for (Map.Entry<SchemaTableName, TableUpdateInfo> entry : updateTables.entrySet()) {
+                tasks.add(() -> {
+                    updateTable(entry.getKey(), entry.getValue());
+                    return null;
+                });
+            }
+
+            this.updateInfos.clear();
+        }
+
+        try {
+            executor.invokeAll(tasks).forEach(MoreFutures::getDone);
+        }
+        catch (InterruptedException e) {
```

Comment:
what about other exceptions?

javadoc states: 
```
If any execution of the task encounters an exception, subsequent executions are suppressed
```

Reply:
you'd need to probably catch ExecutionException and do something about it :)

Reply:
`ExecutionException` is a checked exception. I don't think the above line throws that.